truss/templates/control/control/helpers/inference_server_process_controller.py

```diff
@@ -27,15 +35,18 @@ def start(self):
                 self._inference_server_process_args,
                 env=inf_env,
             )
+
             self._inference_server_started = True
 
     def stop(self):
         if self._inference_server_process is not None:
-            # TODO(pankaj) send sigint wait and then kill
-            poll = self._inference_server_process.poll()
-            if poll is None:
-                self._inference_server_process.kill()
-                self._inference_server_started = False
+            name = " ".join(self._inference_server_process_args)
+
+            for line in os.popen("ps ax | grep '" + name + "' | grep -v grep"):
+                pid = line.split()[0]
+                os.kill(int(pid), signal.SIGKILL)
```

Comment:
Curios why kill by name vs the pid that we already know. I think we likely won't run into this here but ps truncates if the the process name is very long, if we go with this then you may want to check with say a 80 char prefix, rather than the full name. But this is overall more complex, so I'm curios why to do it this way.

Reply:
Ah, yes this is because more recent versions of kserve actually spawn a second daemon process.  So when we want to kill the supervised process, we need to kill both.  I'm open to alternatives to bash execution here, but I couldn't find anything better.  I would have expected the child to be killed off when the popen parent thread was killed, but this seemed unreliable for some reason. Open to discussion here on alternatives.

Reply:
Can we list the child processes and kill them explicitly? Perhaps use psutil? https://stackoverflow.com/questions/3162096/how-do-you-list-all-child-processes-in-python
If we decide to use psutil then we'll need to add it to `truss/templates/control/requirements.txt`.

Reply:
My test cases were from within the docker images so I can verify `ps` is present. I spent some time digging in here and it seemed that `psutil` is a good option but it wasn't readily apparent there was a standard way to do this (could be my lacking knowledge of Popen.  Maybe for this rev we ship with this version and then cleanup as we improve live reload.  Agreed this method isn't preferred.